sklearn/cluster/tests/test_k_means.py

```diff
@@ -771,4 +774,122 @@ def test_x_squared_norms_init_centroids():
 def test_max_iter_error():
 
     km = KMeans(max_iter=-1)
-    assert_raise_message(ValueError, 'Number of iterations should be', km.fit, X)
+    assert_raise_message(ValueError,
+                         'Number of iterations should be', km.fit, X)
+
+
+def test_kmeans_float32_64():
+    km = KMeans(n_init=1, random_state=11)
+
+    # float64 data
+    km.fit(X)
+    # dtype of cluster centers has to be the dtype of the input data
+    assert_equal(km.cluster_centers_.dtype, np.float64)
+    inertia64 = km.inertia_
+    X_new64 = km.transform(km.cluster_centers_)
+    pred64 = km.predict(X[0])
+
+    # float32 data
+    km.fit(np.float32(X))
+    # dtype of cluster centers has to be the dtype of the input data
+    assert_equal(km.cluster_centers_.dtype, np.float32)
+    inertia32 = km.inertia_
+    X_new32 = km.transform(km.cluster_centers_)
+    pred32 = km.predict(X[0])
+
+    # compare arrays with low precision since the difference between
+    # 32 and 64 bit sometimes makes a difference up to the 4th decimal place
+    assert_array_almost_equal(inertia32, inertia64, decimal=4)
+    assert_array_almost_equal(X_new32, X_new64, decimal=4)
```

Comment:
Can you also please add a check that `km.cluster_centers_` in 32 bit is almost equal to its 64 bit counterpart?